src/conways.rs

```diff
@@ -17,10 +19,182 @@ impl ConwaysGrid {
         Self { grid }
     }
 
+    fn modify_cell(&mut self, (row, col): Position, new_state: CellState) {
+        if let Some(grid_row) = self.grid.get_mut(row) {
+            if let Some(cell) = grid_row.get_mut(col) {
+                *cell = new_state;
+            }
+        }
+    }
+
+    pub fn from_alive_cells(alive_cells: &[Position]) -> Self {
+        let mut conways_grid = Self::default();
+        alive_cells
+            .iter()
+            .for_each(|cell| conways_grid.modify_cell(*cell, CellState::Alive));
+        conways_grid
+    }
+
     pub fn next_iteration(&mut self) {
-        for row in self.grid.iter() {
-            for cell in row.iter() {
-                todo!("Implement game rules");
+        let previous_grid = self.grid.clone();
+        for row in 0..ROWS {
+            for col in 0..COLS {
+                self.compute_next_state(&previous_grid, (row, col));
+            }
+        }
+    }
+
+    fn compute_next_state(&mut self, previous_grid: &[Vec<CellState>], (row, col): Position) {
+        let neighbors: Vec<Position> = self.get_neighbor_position((row, col));
+        let alive = Self::get_alive_count(previous_grid, neighbors);
+
+        let grid_row: &Vec<CellState>;
+        let cell: &CellState;
+        if let Some(some_grid_row) = self.grid.get(row) {
+            grid_row = some_grid_row;
+        } else {
+            return;
+        }
+        if let Some(some_cell) = grid_row.get(col) {
+            cell = some_cell;
+        } else {
+            return;
+        }
```

Comment:
Instead of all these lines you can just do
```rust
let Some(grid_row) = self.grid.get(row) else {
            return;
        };
        let Some(cell) = grid_row.get(col) else {
            return;
        };
```

Reply:
Done